pkg/util/metadata/metadata.go

```diff
@@ -205,3 +211,31 @@ func Get(order string) (*Metadata, error) {
 	}
 	return metadataCache, nil
 }
+
+// GetInstanceID from metadata service and config drive
+func (m *Metadata) GetInstanceID(order string) (string, error) {
+	md, err := m.Get(order)
+	if err != nil {
+		return "", err
+	}
+	return md.UUID, nil
+}
+
+// GetAvailabilityZone returns zone from metadata service and config drive
+func (m *Metadata) GetAvailabilityZone(order string) (string, error) {
+	md, err := m.Get(order)
+	if err != nil {
+		return "", err
+	}
+	return md.AvailabilityZone, nil
+}
+
+var IMetadataInterface IMetadata = nil
+
+func GetMetadata() IMetadata {
```

Comment:
GetMetadata is confusing, GetMetadataProvider (as used before) or GetMetadataInstance is better